src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/ThreadPool.Windows.cs

```diff
@@ -180,8 +180,9 @@ private void FinishUnregistering()
             GC.SuppressFinalize(this);
         }
 
-        private void FinishUnregisteringAsync(object waitObject)
+        private void FinishUnregisteringAsync(object? waitObject)
         {
+            Debug.Assert(waitObject != null);
```

Comment:
This is likely causing the CI failures. The code assumes that `safeWaitHandle` can be null below.